tools/openlab/build_core.yaml

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- hosts: all
+  become: yes
+  roles:
+    - role: install-openjdk
+      java_version: '8'
+      with_scala: true
+      scala_version: '2.11.0'
+    - role: install-maven
+      maven_version: '3.2.5'
+
+- hosts: all
+  become: no
+  tasks:
+    - name: Build Flink
+      shell:
+        cmd: |
+          set -xe
+
+          mvn clean install -nsu -Dflink.convergence.phase=install -Pcheck-convergence -Dflink.forkCount=2 \
+          -Dflink.forkCountTestPackage=2 -Dmaven.javadoc.skip=true -B -DskipTests -Dhadoop.version=2.8.3 \
```

Comment:
Since you have described `Add the test script to run unit/intergration test` in this PR, why tests are all skipped by `-DskipTests`?

Reply:
This job name is not correct. I'll fix it.

As I mentioned in the PR commit message. We'll take some steps to add ARM support for Flink. This PR is just the first step:
1. Add the basic build script to ensure the CI system and build job works as expect. The job should be marked as non-voting first, it means the CI test failure won't block Flink PR to be merged.

It just includes build part. So that the test is skipped. We'll add the test part back in the next PR.

Reply:
> Thanks @wangxiyuan for your help, except from my comments, would you please share some documentation of the `theopenlab-ci`? Can we include the ARM built with our own account like https://travis-ci.org/ ?

Sure,
the openlab doc: https://docs.openlabtesting.org/
the CI service which openlab uses:  https://zuul-ci.org/docs/zuul/

And zuul uses `ansible-playbook` for job execution tool, that's what this PR is done.


> Can we include the ARM built with our own account like https://travis-ci.org/ ?

We don't have the account concept in OpenLab, but we have added Flink into OpenLab already. It means that you can do anything you want by editing job related files(The file I created in this PR).  For exampel, adding new kinds of jobs, remove old jobs, change job actions and so on. Or do you want any other request or privilege? Please let me know and I'll response to you if OpenLab supports or not.